### ISSUE TITLE
cross cluster search requests always allowed

### ISSUE THREAD

**Issue by bovy89** [sscarduzio/elasticsearch-readonlyrest-plugin#442]
**ES version**: elasticsearch-6.7.1-1
**ES plugin version**: readonlyrest-1.17.6_es6.7.1.zip

readonlyrest.yml:

```
readonlyrest:

  enable: true

  response_if_req_forbidden: Forbidden by ReadonlyREST ES plugin

  ssl:
    keystore_file: "keystore.jks"
    keystore_pass: readonlyrest
    key_pass: readonlyrest
    allowed_protocols: ["TLSv1.1", "TLSv1.2"]

  access_control_rules:

  - name: "::ADMIN::"
    auth_key_sha256: "<......>"

  - name: "::MONITORING::"
    auth_key_sha256: "<......>"
    actions: ["cluster:monitor/*", "indices:monitor/*"]
    verbosity: error

  - name: "::LOGSTASH::"
    auth_key_sha256: "<......>"
    actions: ["cluster:monitor/main","indices:admin/types/exists","indices:data/read/*","indices:data/write/*","indices:admin/template/*","indices:admin/create"]
    indices: ["logstash-*"]
    verbosity: error

  - name: "::KIBANA-SRV::"
    auth_key_sha256: "<......>"
    verbosity: error

  - name: "::RO_DEV::"
    groups: ["dev"]
    indices: ["mustfail"]
    kibana_access: ro

  - name: "::RO_SYSADMIN::"
    groups: ["sysadmin"]
    indices: [".kibana", "logstash-*"]
    kibana_access: ro

  - name: "::ELKADMIN::"
    groups: ["elkadmin"]
    indices: [".kibana", "logstash-*"]

  users:
  - username: test_dev
    auth_key: test_dev:password
    groups: ["dev"]

  - username: test_sys
    auth_key: test_sys:password
    groups: ["sysadmin"]

  - username: test_admin
    auth_key: test_admin:password
    groups: ["elkadmin"]

```

1. "classic" search (no cross-cluster search syntax) works as expected (FORBIDDEN)

```
# curl -k -u test_dev:password -XGET 'https://test-sys08-centos7.example.com:9200/logstash-system*/_search?pretty'

[2019-05-01T10:36:20,302][INFO ][t.b.r.a.ACL              ] [test-sys08-centos7.example.com] FORBIDDEN by default req={ ID:1271049964-1457807428#554, TYP:SearchRequest, CGR:N/A, USR:test_dev(?), BRS:true, KDX:null, ACT:indices:data/read/search, OA:192.168.95.1, DA:192.168.95.18, IDX:logstash-system*, MET:GET, PTH:/logstash-system*/_search?pretty, CNT:<N/A>, HDR:{Authorization=<OMITTED>, User-Agent=curl/7.38.0, content-length=0, Host=test-sys08-centos7.example.com:9200, Accept=*/*}, HIS:[::ADMIN::->[auth_key_sha256->false]], [::MONITORING::->[auth_key_sha256->false]], [::LOGSTASH::->[auth_key_sha256->false]], [::KIBANA-SRV::->[auth_key_sha256->false]], [::RO_DEV::->[kibana_access->true, indices->false, auth_key->true]], [::RO_SYSADMIN::->[groups->false]], [::ELKADMIN::->[groups->false]] } 
```

2. using cross-cluster search syntax it doesn't works as expected (ALLOW instead of FORBIDDEN)

```
# curl -k -u test_dev:password -XGET 'https://test-sys08-centos7.example.com:9200/cluster_prod:logstash-system*/_search?pretty'

[2019-05-01T10:37:06,603][INFO ][t.b.r.a.ACL              ] [test-sys08-centos7.example.com] ALLOWED by { name: '::RO_DEV::', policy: ALLOW, rules: [groups, kibana_access, indices]} req={ ID:972369622--1830537274#866, TYP:SearchRequest, CGR:N/A, USR:test_dev, BRS:true, KDX:null, ACT:indices:data/read/search, OA:192.168.95.1, DA:192.168.95.18, IDX:cluster_prod:logstash-system*, MET:GET, PTH:/cluster_prod:logstash-system*/_search?pretty, CNT:<N/A>, HDR:{Authorization=<OMITTED>, User-Agent=curl/7.38.0, content-length=0, Host=test-sys08-centos7.example.com:9200, Accept=*/*}, HIS:[::ADMIN::->[auth_key_sha256->false]], [::MONITORING::->[auth_key_sha256->false]], [::LOGSTASH::->[auth_key_sha256->false]], [::KIBANA-SRV::->[auth_key_sha256->false]], [::RO_DEV::->[kibana_access->true, indices->true, auth_key->true]] } 
```

**Comment by coutoPL** [comment]
@bovy89 thanks for report. Yes, this is a bug. I was able to reproduce it. Will notify you when we fix

**Comment by coutoPL** [comment]
@bovy89 the bug is fixed in v1.18.0.
